express/blocks/fullscreen-marquee/fullscreen-marquee.js

```diff
@@ -0,0 +1,175 @@
+
+      thumbnailContainer.append(thumbnail);
+      app.append(thumbnailContainer);
+
+      content.addEventListener('loadedmetadata', () => {
+        const framesContainer = createTag('div', { class: 'fullscreen-marquee-app-frames-container' });
+        function createFrame(current, total) {
+          const frame = createTag('video', { src: `${content.currentSrc}#t=${current}` });
+          framesContainer.append(frame);
+
+          frame.addEventListener('loadedmetadata', () => {
+            frame.style.opacity = '1';
+
+            if (current < total) {
+              const newFrameCount = current + 1;
+              createFrame(newFrameCount, total);
+            }
+          });
+        }
+
+        createFrame(1, 10);
+        app.append(framesContainer);
+      });
+
+      thumbnail.addEventListener('loadedmetadata', () => {
+        thumbnail.currentTime = Math.floor(thumbnail.duration) / 2 || 0;
+        thumbnail.pause();
+      });
+    }
+  } else {
+    variant = 'image';
+  }
+
+  await import(`${getLibs()}/features/placeholders.js`).then(async (mod) => {
+    const imgSrc = await mod.replaceKey(`fullscreen-marquee-desktop-${variant}-app`, getConfig());
+    appImage = createOptimizedPicture(imgSrc);
+
+    const editorSrc = await mod.replaceKey(`fullscreen-marquee-desktop-${variant}-editor`, getConfig());
+    editor = createOptimizedPicture(editorSrc);
+
+    appImage.classList.add('fullscreen-marquee-app-image');
+    return mod.replaceKey();
+  });
+
+  editor.classList.add('fullscreen-marquee-app-editor');
+  content.classList.add('fullscreen-marquee-app-content');
+
+  window.addEventListener('mousemove', (e) => {
+    const rotateX = ((e.clientX * 10) / (window.innerWidth / 2) - 10);
+    const rotateY = -((e.clientY * 10) / (window.innerHeight / 2) - 10);
+
+    app.style.transform = `rotateX(${rotateY}deg) rotateY(${rotateX}deg) translate3d(${rotateX}px, 0px, 0px)`;
+    appBackground.style.transform = `rotateX(${rotateY}deg) rotateY(${rotateX}deg) translate3d(${rotateX}px, 0px, -50px)`;
+  }, { passive: true });
+
+  contentContainer.append(content);
+  app.append(appImage);
+  app.append(contentContainer);
+  appFrame.append(app);
+  appFrame.append(appBackground);
+  app.append(editor);
+  addMarqueeCenterCTA(block, appFrame);
+  return appFrame;
+}
+
+export default async function decorate(block) {
+  const rows = Array.from(block.children);
+  const heading = rows[0] ? rows[0].querySelector('div') : null;
+  const background = rows[2] ? rows[2].querySelector('picture') : null;
+  let content = rows[1] ?? null;
+
+  decorateButtonsDeprecated(block);
+
+  block.innerHTML = '';
+
+  if (content) {
+    content = buildContent(content);
+  }
+
+  if (background) {
+    block.classList.add('has-background');
+    block.append(buildBackground(block, background));
+  }
+
+  if (heading) {
+    if (getConfig().locale.region === 'jp') heading.querySelector('h1').classList.add('budoux');
```

Comment:
really tiny nit: it would probably be slightly better to check like so - `config.locale?.ietf === 'ja-JP'`, just in case we ever were to ever offer the japanese site in english as well